field_encrypt.module

```diff
@@ -131,44 +124,115 @@ function field_encrypt_form_field_add_form_builder($entity_type, \Drupal\field\E
     /**
      * @var $field_encrypt_process_entities \Drupal\field_encrypt\FieldEncryptProcessEntities
      */
-    $field_encrypt_process_entities = \Drupal::service('field_encrypt.process_entities');
-    if ($form_encryption === 1) {
-      $field_encrypt_process_entities->encryptStoredField($field_entity_type, $field_name);
-    }
-    elseif ($form_encryption === 0) {
-      $field_encrypt_process_entities->decryptStoredField($field_entity_type, $field_name);
-    }
+    // @TODO: refactor logic for re-encrypting existing fields - currently broken, so commented out.
+    //    $field_encrypt_process_entities = \Drupal::service('field_encrypt.process_entities');
+    //    if ($form_encryption === 1) {
+    //      $field_encrypt_process_entities->encryptStoredField($field_entity_type, $field_name);
+    //    }
+    //    elseif ($form_encryption === 0) {
+    //      $field_encrypt_process_entities->decryptStoredField($field_entity_type, $field_name);
+    //    }
   }
 }
 
 /**
- * Encrypt fields before they are saved.
+ * Implements hook_entity_insert().
+ *
+ * Encrypts the entity after being saved for the first time.
+ */
+function field_encrypt_entity_insert(Drupal\Core\Entity\EntityInterface $entity) {
+  // Save the entity again, but now encrypted.
+  // @TODO: check if there's not a cleaner way to do this.
+  $saved_entity = entity_load($entity->getEntityTypeId(), $entity->id());
```

Comment:
The $entity variable passed as parameter does not have an ID assigned yet.
So $entity->id() returns no result, which prevents us from saving EncryptedFieldValue entities that reference that entity ID.

Perhaps we can get the entity ID in another way, but I haven't yet figured out how.